src/apps/common/submodules/portWizard/portWizard.js

```diff
@@ -318,64 +318,420 @@ define(function(require) {
 				portRequestId = _.get(args, 'data.portRequestId'),
 				i18n = self.i18n.active().commonApp.portWizard,
 				i18nSteps = i18n.steps,
-				stepNames = self.appFlags.portWizard.stepNames,
-				$container,
-				callback;
+				stepNames = self.appFlags.portWizard.stepNames;
 
-			if (args.container instanceof jQuery) {
-				$container = args.container;
-				callback = globalCallback;
-			} else {
-				var modal = monster.ui.fullScreenModal(null, {
-					inverseBg: true
-				});
-				$container = $('.core-absolute').find('#' + modal.getId() + ' .modal-content');
-				callback = function() {
-					modal.close();
+			monster.waterfall([
+				function(waterfallCallback) {
+					// Clean store, in case it was not empty, to avoid using old data
+					self.portWizardSet({});
 
-					globalCallback();
-				};
-			}
+					// Get container and main callback function
+					if (args.container instanceof jQuery) {
+						return waterfallCallback(null, args.container, globalCallback);
+					}
 
-			self.portWizardSet({
-				accountId: accountId,
-				globalCallback: callback
-			});
+					var modal = monster.ui.fullScreenModal(null, {
+							inverseBg: true
+						}),
+						$container = $('.core-absolute').find('#' + modal.getId() + ' .modal-content'),
+						callback = function() {
+							modal.close();
 
-			monster.pub('common.navigationWizard.render', {
-				thisArg: self,
-				controlId: 'port_wizard_control',
-				data: {
-					portRequestId: portRequestId,
-					nameAndNumbers: {
-						numbersToPort: {
-							type: 'local'
+							globalCallback();
+						};
+
+					waterfallCallback(null, $container, callback);
```

Comment:
I'm not sure that I understand this correctly. I think that the closing handler cannot be passed as is, because the `globalCallback` should be executed after it too, so it is wrapped within a new function that calls both of them.

Reply:
> I think that the closing handler cannot be passed as is, because the globalCallback should be executed after it too

Since `globalCallback` is a global variable in this block, it is accessible in any task of your waterfall flow. Meaning that, instead of creating your wrapping `callback` function in that task, you could just pass the modal close handler and construct that `callback` in the next one:

```js
					waterfallCallback(null, $container, modal);
				},
				function($container, closeModalHandler, waterfallCallback) {
				    self.portWizardSet({
				          accountId: accountId,
				          globalCallback: function() {
				              modal.close();
				              globalCallback();
				          }
				    });
				}
```
Or you could pass down `modal.close` directly.